File: src/main/java/com/cloudogu/gitops/infrastructure/git/providers/scmmanager/api/AuthorizationInterceptor.java
package com.cloudogu.gitops.infrastructure.git.providers.scmmanager.api;

import java.io.IOException;
import lombok.RequiredArgsConstructor;
import okhttp3.Credentials;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
import org.jetbrains.annotations.NotNull;

/** OkHttp interceptor that adds HTTP basic auth credentials to every SCM-Manager request. */
@RequiredArgsConstructor
public class AuthorizationInterceptor implements Interceptor {
private final String username;
private final String password;

@Override
public Response intercept(@NotNull Chain chain) throws IOException {
	Request newRequest =
		chain
			.request()
			.newBuilder()
			.header("Authorization", Credentials.basic(username, password))
			.build();

	return chain.proceed(newRequest);
}
}

File: src/main/kotlin/com/cloudogu/gitops/infrastructure/git/providers/scmmanager/api/AuthorizationInterceptor.kt
package com.cloudogu.gitops.infrastructure.git.providers.scmmanager.api

import okhttp3.Credentials
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class AuthorizationInterceptor(
    private val username: String,
    private val password: String
) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val newRequest = chain.request().newBuilder()
            .header("Authorization", Credentials.basic(username, password))
            .build()

        return chain.proceed(newRequest)
    }
}
